### PR TITLE
Backoff: Change the algorithm from a class to a function

### PR DESCRIPTION
**Pull request by Geod24**
Using a class for this is quite overkill, and seems to only have been
done to make the utility testable, which can be achieved just as well
by using a template parameter.

### CODE REVIEW

File: source/agora/utils/Backoff.d [hunk 1]
```diff
@@ -19,68 +19,55 @@ import libsodium.randombytes;
...
+    ```
+    sleep_time = random(0, min(max_delay, base * (2 ^^ attempt));
+    ```
 
-    ***************************************************************************/
+    Params:
+        RandomFunc = A function that takes a `uint` as an input and returns a
+                     value between `0` and the provided value, uniformly
+        attempt = the attempt number (can be 0, which will result in 0)
+        base    =
```
Comment (hewison-chris) [inline source/agora/utils/Backoff.d]: ?

File: source/agora/utils/Backoff.d [hunk 2]
```diff
@@ -19,68 +19,55 @@ import libsodium.randombytes;
...
-    ***************************************************************************/
+*******************************************************************************/
 
-    protected uint getRandom (uint value)
-    {
-        return () @trusted { return randombytes_uniform(value); }();
-    }
+public uint getRandom (uint value) @safe nothrow @nogc
+{
+    return () @trusted { return randombytes_uniform(value); }();
```
Comment (omerfirmak) [inline source/agora/utils/Backoff.d:71]: I would not call this "jitter", maybe we should take a look at this later.

**Comment by Geod24** [comment]
Fixed documentation

**Review by linked0 — APPROVED** [review]
LGTM

**Pull request merged**